src/main/java/appeng/fluids/parts/PartFluidAnnihilationPlane.java

```diff
@@ -0,0 +1,322 @@
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( e ) ), this.getSide() ) )
+			{
+				maxX = 16;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( u.getOpposite() ) ), this.getSide() ) )
+			{
+				minY = 0;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( e ) ), this.getSide() ) )
+			{
+				maxY = 16;
+			}
+		}
+		
+		bch.addBox( 5, 5, 14, 11, 11, 15 );
+		bch.addBox( minX, minY, 15, maxX, maxY, 16 );
+	}
+
+	public PlaneConnections getConnections()
+	{
+
+		final EnumFacing facingRight, facingUp;
+		AEPartLocation location = this.getSide();
+		switch( location )
+		{
+			case UP:
+				facingRight = EnumFacing.EAST;
+				facingUp = EnumFacing.NORTH;
+				break;
+			case DOWN:
+				facingRight = EnumFacing.WEST;
+				facingUp = EnumFacing.NORTH;
+				break;
+			case NORTH:
+				facingRight = EnumFacing.WEST;
+				facingUp = EnumFacing.UP;
+				break;
+			case SOUTH:
+				facingRight = EnumFacing.EAST;
+				facingUp = EnumFacing.UP;
+				break;
+			case WEST:
+				facingRight = EnumFacing.SOUTH;
+				facingUp = EnumFacing.UP;
+				break;
+			case EAST:
+				facingRight = EnumFacing.NORTH;
+				facingUp = EnumFacing.UP;
+				break;
+			default:
+			case INTERNAL:
+				return PlaneConnections.of( false, false, false, false );
+		}
+
+		boolean left = false, right = false, down = false, up = false;
+
+		final IPartHost host = this.getHost();
+		if( host != null )
+		{
+			final TileEntity te = host.getTile();
+
+			final BlockPos pos = te.getPos();
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( facingRight.getOpposite() ) ), this.getSide() ) )
+			{
+				left = true;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( facingRight ) ), this.getSide() ) )
+			{
+				right = true;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( facingUp.getOpposite() ) ), this.getSide() ) )
+			{
+				down = true;
+			}
+
+			if( this.isAnnihilationPlane( te.getWorld().getTileEntity( pos.offset( facingUp ) ), this.getSide() ) )
+			{
+				up = true;
+			}
+		}
+
+		return PlaneConnections.of( up, right, down, left );
+	}
+
+	@Override
+	public void onNeighborChanged( IBlockAccess w, BlockPos pos, BlockPos neighbor )
+	{
+		if( pos.offset( this.getSide().getFacing() ).equals( neighbor ) )
+		{
+			this.refresh();
+		}
+	}
+
+	@Override
+	public float getCableConnectionLength( AECableType cable )
+	{
+		return 1;
+	}
+
+	private boolean isAnnihilationPlane( final TileEntity blockTileEntity, final AEPartLocation side )
+	{
+		if( blockTileEntity instanceof IPartHost )
+		{
+			final IPart p = ( (IPartHost) blockTileEntity ).getPart( side );
+			return p != null && p.getClass() == this.getClass();
+		}
+		return false;
+	}
+
+	private void refresh()
+	{
+		try
+		{
+			this.getProxy().getTick().alertDevice( this.getProxy().getNode() );
+		}
+		catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)